handler/src/main/java/io/netty/handler/timeout/IdleStateHandler.java

```diff
@@ -355,15 +377,72 @@ protected IdleStateEvent newIdleStateEvent(IdleState state, boolean first) {
             case WRITER_IDLE:
                 return first ? IdleStateEvent.FIRST_WRITER_IDLE_STATE_EVENT : IdleStateEvent.WRITER_IDLE_STATE_EVENT;
             default:
-                throw new Error();
+                throw new IllegalArgumentException("Unhandled: state=" + state + ", first=" + first);
+        }
+    }
+
+    /**
+     * @see #hasOutputChanged(ChannelHandlerContext, boolean)
+     */
+    private void initOutputChanged(ChannelHandlerContext ctx) {
+        if (observeOutput) {
+            Channel channel = ctx.channel();
+            Unsafe unsafe = channel.unsafe();
+            ChannelOutboundBuffer buf = unsafe.outboundBuffer();
+
+            lastMessageHashCode = System.identityHashCode(buf.current());
```

Comment:
maybe not when this method is currently called but it is possible the `ChannelOutboundBuffer` may be `null` (e.g. channel is closed) .. consider protecting against this or adding a comment as to why it isn't necessary.